versioned_docs/version-2.x/magic-button/preview-mode.mdx

```diff
@@ -0,0 +1,125 @@
+---
+sidebar_position: 3
+title: Preview Mode
+description:
+  Get a sneak peek of your content's appearance and make necessary changes for
+  an optimal user experience.
+---
+
+import BrowserWindow from "@site/src/components/BrowserWindow";
+import SinceVersion from "@site/src/components/SinceVersion";
+
+<SinceVersion tag="2.25" />
+
+**Preview Mode** is an integral feature of Magic Button, specifically designed
+to enable content creators to preview their drafts from the CMS before they go
+live.
+
+## How to activate Preview Mode
+
+### Using the Magic Button
+
+Activate the **Preview Mode** directly from the Magic Button. Click on the Magic
+Button and select the Preview Mode option.
+
+<div style={{ width: 600, maxWidth: "100%" }}>
+  <BrowserWindow noPadding>
+    <img
+      src={require("./assets/enable-preview.gif").default}
+      width="100%"
+      height="100%"
+      style={{ objectFit: "contain", marginBottom: -10 }}
+    />
+  </BrowserWindow>
+</div>
+
+### Programmatically
+
+Alternatively, **Preview Mode** can be activated programmatically by leveraging
+the
+[ContributionModeLoader](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/7a2aee7e11c7bd45cfc3a5a940407f9e220e66f4/src/server/modules/Front-Commerce/Contribution-Mode/contribution-mode/loaders/ContributionModeLoader.js#L57-60).
+There are two methods for this:
+
+### `setPreviewMode`
+
+The **setPreviewMode** method is typically used for safe activation of Preview
+Mode. If Contribution Mode is not enabled, Preview Mode will not be activated.
+
+This method is exposed in the
+[Contribution GraphQL schema](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/7a2aee7e11c7bd45cfc3a5a940407f9e220e66f4/src/server/modules/Front-Commerce/Contribution-Mode/contribution-mode/schema.gql)
+
+```graphql
+mutation EnablePreviewMode($enable: Boolean) {
+  contribution(input: $enable) {
+    setPreviewMode
+  }
+}
+```
+
+You can also call this method directly from the loader:
+
+```js
+export default {
+  namespace: "MyModule",
+  dependencies: ["Front-Commerce/Contribution-Mode"],
+  resolvers: {
+    Mutation: {
+      myQuery: async (root, args, context, info) => {
```

Comment:
same feedback than for contribution mode: a more realistic example should be used